sympy/core/add.py

```diff
@@ -679,8 +679,12 @@ def _eval_is_imaginary(self):
                     return
             elif a.is_imaginary:
                 im_I.append(a*S.ImaginaryUnit)
-            elif (S.ImaginaryUnit*a).is_extended_real:
-                im_I.append(a*S.ImaginaryUnit)
+            elif a.is_Mul and S.ImaginaryUnit in a.args:
+                coeff, ai = a.as_coeff_mul(S.ImaginaryUnit)
+                if ai == (S.ImaginaryUnit,) and coeff.is_extended_real:
+                    im_I.append(a*S.ImaginaryUnit)
```

Comment:
This general pattern does still create an expression though, right?

Reply:
It does. I want here with @smichr's suggestion that reduces the impact of the change slightly.

Thinking about it now `im_I.append(Mul(*ai, evaluate=False))` would be better so that we know that we produce a smaller Mul without depending on evaluation.

Reply:
Rather `im_I.append(-coeff)`. If `a = I*coeff` then `I*a = -coeff`.

Reply:
I've just changed this to `-coeff`.